staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/kms/google/gce.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package google
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	gcfg "gopkg.in/gcfg.v1"
+
+	"github.com/golang/glog"
+
+	"cloud.google.com/go/compute/metadata"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	cloudkms "google.golang.org/api/cloudkms/v1"
+	compute "google.golang.org/api/compute/v1"
+)
+
+type GoogleCloudkmsService struct {
+	CloudkmsService *cloudkms.Service
+	ProjectID       string
+}
+
+// Has to be the same as the configuration available in the GCE cloudprovider.
+// Currently at k8s.io/kubernetes/pkg/cloudprovider/providers/gce.
+// Although only the first 3 fields are needed, gcfg does not parse the configuration
+// if there are extra elements in the actual file. Which is why, we need to keep all the
+// fields here.
+type cloudConfig struct {
+	Global struct {
+		TokenURL           string   `gcfg:"token-url"`
```

Comment:
There is `gcfg.FataOnly` https://godoc.org/gopkg.in/gcfg.v1#FatalOnly. Wouldn't that hep to strip this down?